### PR TITLE
Add RPC `merkle.treeLeaves` to `Merkle` pallet

### PR DESCRIPTION
**Pull request by shekohex**

### CODE REVIEW

**Comment by shekohex** [comment]
@filiplazovic could you please help me fixing these tests to compile? we renamed things here (Group -> Tree) but don't know where all of them.

File: runtime/src/lib.rs [hunk 1]
```diff
@@ -813,6 +815,17 @@ impl_runtime_apis! {
 		}
 	}
 
+	impl merkle::MerkleApi<Block> for Runtime {
+		fn get_leaves(tree_id: u32, index: u32) -> Option<ScalarData> {
```
Comment (drewstone) [inline runtime/src/lib.rs:819]: Should we call this `get_leaf`?
```suggestion
		fn get_leaf(tree_id: u32, index: u32) -> Option<ScalarData> {
```

File: pallets/merkle/rpc/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+		if to - from >= 512 {
+			return Err(Error {
+				code: ErrorCode::ServerError(1512), // Too many leaves
+				message: "TooManyLeaves".into(),
+				data: Some("MaxRange512".into()),
+			});
+		}
+		let leaves = (from..to)
+			.into_iter()
+			.map(|i| api.get_leaves(&at, tree_id, i as u32)) // Result<Option<ScalarData>>
```
Comment (drewstone) [inline pallets/merkle/rpc/src/lib.rs:68]:
```suggestion
			.map(|i| api.get_leaf(&at, tree_id, i as u32)) // Result<Option<ScalarData>>
```

File: pallets/merkle/src/lib.rs [hunk 3]
```diff
@@ -429,6 +428,13 @@ pub mod pallet {
 	}
 }
 
+sp_api::decl_runtime_apis! {
+	pub trait MerkleApi {
+		/// Get leaves of tree id.
+		fn get_leaves(tree_id: u32, index: u32) -> Option<ScalarData>;
```
Comment (drewstone) [inline pallets/merkle/src/lib.rs:434]:
```suggestion
		fn get_leaf(tree_id: u32, index: u32) -> Option<ScalarData>;
```

**Review by drewstone — CHANGES_REQUESTED** [review]
Did you also try using the `iter_prefix_values` to see if its faster than querying one at a time?

File: pallets/merkle/src/lib.rs [hunk 4]
```diff
@@ -429,6 +428,13 @@ pub mod pallet {
 	}
 }
 
+sp_api::decl_runtime_apis! {
+	pub trait MerkleApi {
+		/// Get leaves of tree id.
```
Comment (drewstone) [inline pallets/merkle/src/lib.rs:433]:
```suggestion
		/// Get the leaf of tree id at a given index.
```

**Review by drewstone — CHANGES_REQUESTED** [review]

**Pull request merged**